crates/criticaltrust/src/signatures/keychain.rs

```diff
@@ -158,6 +193,67 @@ mod tests {
         assert!(matches!(keychain.load(&other), Err(Error::UnsupportedKey)));
     }
 
+    impl Signable for String {
+        const SIGNED_BY_ROLE: KeyRole = KeyRole::Packages;
+    }
+
+    impl Signable for Vec<u8> {
+        const SIGNED_BY_ROLE: KeyRole = KeyRole::Revocation;
+    }
+
+    #[test]
+    fn test_load_all_revoked_content_empty() {
+        // Test `load_all` method with RevocationInfo but empty list.
+        let root = generate_key(KeyRole::Root);
+        let revocation = generate_trusted_key(KeyRole::Revocation, &root);
+
+        let revoked_content = RevocationInfo::new(vec![], datetime!(2400-10-10 00:00 UTC));
+        let mut signed_revoked_content = SignedPayload::new(&revoked_content).unwrap();
+        signed_revoked_content.add_signature(&revocation.0).unwrap();
+
+        let mut keychain = Keychain::new(root.public()).unwrap();
+
+        let keys_manifest = KeysManifest {
+            version: ManifestVersion,
+            keys: vec![revocation.1],
+            revoked_signatures: signed_revoked_content,
+        };
+
+        keychain.load_all(&keys_manifest).unwrap();
+        assert_eq!(
+            keychain.revocation_info.unwrap().expires_at,
+            datetime!(2400-10-10 00:00 UTC)
+        )
+    }
+
+    #[test]
+    fn test_load_all_revoked_content_one_item() {
```

Comment:
Perhaps an added test to answer "What is the expected behaviour if the revoked content contains a hash that is signed by something that is not the revocation key?" would be useful? (If we somehow stop checking that it would be a security issue)